pkg/cri/resource-manager/cache/cache.go

```diff
@@ -901,7 +910,7 @@ func (cch *cache) LookupPod(id string) (Pod, bool) {
 }
 
 // Insert a container into the cache.
-func (cch *cache) InsertContainer(msg interface{}) (Container, error) {
+func (cch *cache) InsertContainer(msg interface{}, status *cri.ContainerStatusResponse) (Container, error) {
```

Comment:
nit/note: this function prototype just looks really cranky

Could we e.g. split this into two separate functions, one called from outside of the `cache´ package:

```
func (cch *cache) InsertContainer(msg *cri.CreateContainerRequest) (Container, error) {
	c := &container{
		cache: cch,
	}
	if err := c.fromCreateRequest(msg); err != nil {
		return nil, cacheError("failed to insert container %s: %v", c.CacheID, err)
	}
	return cch.insertContainer(c)
}
```

...and one internal function that would contain most of the logic here:

```
func (cch *cache) insertContainer(c *container) (Container, error) {
   ...
}
```

We would then move the logic from `case *cri.Container: ` down into the `RefreshContainers()` method.

WDYT? Could be done in a separate pr, too